File: 백준/Silver/1010. 다리 놓기/다리 놓기.py
import math

def factorial(n):
    return math.factorial(n)

def combination(n, r):
    if r == 0 or n == r:
        return 1
    return factorial(n) // (factorial(r) * factorial(n - r))

n = []
m = []

number = int(input())

for i in range(number):
    n_var, m_var = map(int, input().split())
    n.append(n_var)
    m.append(m_var)

for i in range(number):
    print(combination(m[i], n[i]))

File: 백준/Silver/1010. 다리 놓기/다리 놓기.java
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.io.IOException;
import java.math.BigInteger;
import java.util.StringTokenizer;

public class Main {
    
    // 팩토리얼을 계산하는 함수
    static BigInteger factorial(int n) {
        BigInteger result = BigInteger.ONE;
        for (int i = 2; i <= n; i++) {
            result = result.multiply(BigInteger.valueOf(i));
        }
        return result;
    }

    // 조합을 계산하는 함수
    static BigInteger combination(int n, int r) {
        if (r == 0 || n == r) {
            return BigInteger.ONE;
        }
        return factorial(n).divide(factorial(r).multiply(factorial(n - r)));
    }

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int T = Integer.parseInt(br.readLine());
        
        for (int t = 0; t < T; t++) {
            StringTokenizer st = new StringTokenizer(br.readLine());
            int N = Integer.parseInt(st.nextToken());
            int M = Integer.parseInt(st.nextToken());
            System.out.println(combination(M, N));
        }
    }
}
